src/videodownlinkbandwidthpolicy/VideoPriorityBasedPolicy.ts

```diff
@@ -336,29 +304,13 @@ export default class VideoPriorityBasedPolicy implements VideoDownlinkBandwidthP
     };
     rates.targetDownlinkBitrate = this.determineTargetRate();
 
-    // calculate subscribe delay based on bandwidth increasing/decreasing
     const numberOfParticipants = this.subscribedReceiveSet.size();
-    const prevEstimated = this.prevDownlinkStats.bandwidthEstimateKbps;
-    const currEstimated = this.downlinkStats.bandwidthEstimateKbps;
-
-    if (currEstimated > prevEstimated) {
-      // if bw increases, we use recovery delay
-      this.timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
-        NetworkEvent.Increase,
-        numberOfParticipants
-      );
-    } else if (currEstimated < prevEstimated) {
-      // if bw decreases, we use response delay
-      this.timeBeforeAllowSubscribeMs = this.getSubscribeDelay(
-        NetworkEvent.Decrease,
-        numberOfParticipants
-      );
-    }
+    const currentEstimated = this.downlinkStats.bandwidthEstimateKbps;
 
     // If no major changes then don't allow subscribes for the allowed amount of time
     if (
       noMajorChange &&
-      Date.now() - this.lastSubscribeTimestamp < this.timeBeforeAllowSubscribeMs
+      !this.videoPriorityBasedPolicyConfig.allowSubscribe(numberOfParticipants, currentEstimated)
```

Comment:
Cool! Any time we are able to abstract away a good chunk of code and state (technically didn't *remove* state but we probably would have needed to add it) which results in a diff like this is a good win. 

As a first time reader, one now doesn't need to clutter their attempts to initially understand the `VideoPriorityBasedPolicy` with the supplementary adaption delay configuration (which they can save for another time :).